File: Assets/Script/NetworkClient/proto/Messages.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
//---------------System Message----------------
public class MsgPing : MsgBase
{
    public MsgPing() { protoName = "MsgPing"; }
}

public class MsgPong : MsgBase
{
    public MsgPong() { protoName = "MsgPong"; }
}
//---------------Agent Messages-----------------
public class TransitionMsg : MsgBase 
{
    public TransitionMsg() { protoName = "TransitionMsg"; }
    public float[] state;
    public float[] action;
    public float[] reward;
    public float[] next_state;
    public bool trancated_flag;
}

public class SampleActionRequestMsg : MsgBase
{
    public SampleActionRequestMsg() { protoName = "SampleActionRequestMsg"; }
    public int agent_id;
    public float[] state;
}

public class SampleActionResponseMsg : MsgBase
{
    public SampleActionResponseMsg() { protoName = "SampleActionResponseMsg"; }
    public int agent_id;
    public float[] action;
}

public class EpisodicRewardMsg : MsgBase
{
    public EpisodicRewardMsg() { protoName = "EpisodicRewardMsg"; }
    public float reward;
}

//-----------------Motion Capture Message---------------------
public class MotionStateMsg : MsgBase
{
    public MotionStateMsg() {  protoName = "MotionStateMsg"; }
    public int step;
    public float[] dof_pos_RH;
}
//--------------Test Message----------
public class MsgTest : MsgBase
{
    public MsgTest() { protoName = "MsgTest"; }
    public string str;
    public int num;
}

File: Assets/PythonCode/NetworkServer/proto/Messages.py
import os
import sys
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from net.MsgBase import MsgBase
#----------------System Message----------------
class MsgPing(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "MsgPing"

class MsgPong(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "MsgPong"
#---------------Motion Capture Message----------------
class MotionStateMsg(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "MotionStateMsg"
        self.dof_pos_RH = []  
#---------------Agent Message----------------
class TransitionMsg(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "TransitionMsg"
        self.state = []
        self.action = []
        self.reward = []
        self.next_state = []
        self.trancated_flag = False 
class SampleActionRequestMsg(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "SampleActionRequestMsg"
        self.agent_id = -1
        self.state = []
class SampleActionResponseMsg(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "SampleActionResponseMsg"
        self.agent_id = -1
        self.action = []
class EpisodicRewardMsg(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "EpisodicRewardMsg"
        self.reward = 0
#---------------Test Message----------------
class MyMessage(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "MyMessage"
        self.content = ""
class MsgTest(MsgBase):
    def __init__(self):
        super().__init__()
        self.protoName = "MsgTest"
        self.str = ""
        self.num = 0